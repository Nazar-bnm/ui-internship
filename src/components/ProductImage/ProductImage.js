import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import Carousel from '../Carousel';
import ProductImageItem from './ProductImageItem';

import './ProductImage.scss';

export const CN = 'product-image';

const visibleNumOfSlides = {
  desktop: 3,
  tablet: 3,
  mobile: 3
};

class ProductImage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      selectedImage: '',
      errored: false
    };

    this.handleImageError = this.handleImageError.bind(this);
  }

  getBigImageSrc() {
    const { selectedImage } = this.state;
    const { images } = this.props;
    if (selectedImage === '') {
      return images[0].src;
    }

    return selectedImage;
  }

  clickHandler = ({ target }) => {
    const selectedImage = target.dataset.src;

    this.setState({
      selectedImage
    });
  };

  handleImageError() {
    const { errored } = this.state;
    if (!errored) {
      this.setState({
        selectedImage: 'src/assets/img/productListPage/placeholder-image.jpeg',
        errored: true
      });
    }
  }

  renderSmallImages() {
    const {
      images,
      verticalCarousel
    } = this.props;
    const { selectedImage } = this.state;

    const smallImages = images.map((el) => {
      const { src } = el;
      const isSelected = selectedImage === src;

      return (
        <ProductImageItem
          keyForChild={src}
          key={src}
          src={src}
          selectedImage
          isSelected={isSelected}
          clickHandler={this.clickHandler}
        />
      );
    });

    return (
      <Carousel
        className="carousel-image"
        visibleNumOfSlides={visibleNumOfSlides}
        vertical={verticalCarousel}
      >
        {smallImages}
      </Carousel>
    );
  }

  render() {
    const { selectedImage } = this.state;
    const { className } = this.props;

    return (
      <div className={cx(`${CN}__container`, className)}>
        {this.renderSmallImages(selectedImage)}
        <div className={`${CN}__big-image-container`} type="submit">
          <img
            alt="product"
            className={CN}
            src={this.getBigImageSrc()}
            onError={this.handleImageError}
          />
        </div>
      </div>
    );
  }
}

ProductImage.propTypes = {
  images: PropTypes.arrayOf(
    PropTypes.shape({
      src: PropTypes.string
    })
  ),
  className: PropTypes.string
};

ProductImage.defaultProps = {
  images: [],
  className: ''
};

export default ProductImage;
