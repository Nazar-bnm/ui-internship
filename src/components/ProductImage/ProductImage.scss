@import "../../scss/global_styles";

.product-image {
  &__container {
    @include flex();
    padding: 0;
    max-width: 350px;
  }

  &__big-image-container {
    @include flex($justify: center);
    width: 100%;
    position: relative;

    & img {
      width: 100%;
    }
  }

  &__small-images-container {
    @include flex($wrap: nowrap);
    padding: 10px 0;
    width: 100%;
  }

  &__small-image {
    @include flex();
    position: relative;

    &:nth-child(2) {
      margin: 0 5px;
    }
  }

  & {
    min-width: 50px;
    max-width: 100%;
    height: auto;
  }

  &__image-on-hover {
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    opacity: 0;
    transition: .5s ease;
    background-color: rgba(244, 162, 54, 0.7);
    overflow: hidden;
  
    &:hover {
      opacity: 1;
    }
  }

  &__selected-image {
    border: 2px solid rgba(244, 162, 54, 1);;
  }
}

@media only screen and (min-width: 780px) {
  .product-image__container {
    margin-right: 20px;
  }
}
