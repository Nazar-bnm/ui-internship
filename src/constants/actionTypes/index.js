// Counter
export const INCREMENT = 'INCREMENT';
export const DECREMENT = 'DECREMENT';
// Wish list
export const ADD_TO_WISHLIST = 'ADD_TO_WISHLIST';
export const REMOVE_FROM_WISHLIST = 'REMOVE_FROM_WISHLIST';
export const CHANGE_QUANTITY = 'INCREMENT_QUANTITY';
export const REMOVE_ITEM_FROM_CART = 'REMOVE_ITEM_FROM_CART';
// Filters
export const CHECK_CHECKBOX = 'CHECK_CHECKBOX';
export const RESET_FILTERS = 'RESET_FILTERS';
export const SET_PRICE_RANGE = 'SET_PRICE_RANGE';
// Products pagination
export const FETCH_PRODUCTS_PENDING = 'FETCH_PRODUCTS_PENDING';
export const FETCH_PRODUCTS_SUCCESS = 'FETCH_PRODUCTS_SUCCESS';
export const FETCH_PRODUCTS_ERROR = 'FETCH_PRODUCTS_ERROR';
export const CHANGE_ITEMS_NUMBER_ON_PAGE = 'CHANGE_ITEMS_NUMBER_ON_PAGE';
export const CHANGE_SORTING_ORDER = 'CHANGE_SORTING_ORDER';
export const CHANGE_ORDER_TYPE = 'CHANGE_ORDER_TYPE';
export const GET_PRODUCTS_SUCCESS = 'GET_PRODUCTS_SUCCESS';
export const GET_PRODUCTS_ERROR = 'GET_PRODUCTS_ERROR';
export const NOTIFICATION_SHOW = 'NOTIFICATION_SHOW';
export const NOTIFICATION_HIDE = 'NOTIFICATION_HIDE';
export const ADD_TO_CART = 'ADD_TO_CART';
export const NOTIFICATION_ITEM_ADD_TO_LIST = 'NOTIFICATION_ITEM_ADD_TO_LIST';
export const NOTIFICATION_ITEM_REMOVE_FROM_LIST = 'NOTIFICATION_ITEM_REMOVE_FROM_LIST';

// Blog page
export const GET_BLOG_ITEMS_SUCCESS = 'GET_BLOG_ITEMS_SUCCESS';
export const GET_BLOG_ITEMS_ERROR = 'GET_BLOG_ITEMS_ERROR';

// Cart
export const SET_TOTAL_COUNT = 'SET_TOTAL_COUNT';
