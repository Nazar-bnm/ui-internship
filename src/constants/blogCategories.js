export const blogCategories = ['celebrity style', 'fashion shows', 'causal events', 'shopping'];
