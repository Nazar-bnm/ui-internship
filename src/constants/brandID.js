export const brandIDEnum = {
  boss: '5daf1650a3f2fe001790b09b',
  donnakaran: '5daf1729a3f2fe001790b09c',
  gucci: '5daf17a6a3f2fe001790b09d',
  baldinini: '5daf17fba3f2fe001790b09e',
  dior: '5daf1850a3f2fe001790b09f',
  emporioarmani: '5daf191ba3f2fe001790b0a0',
  calvinklein: '5daf1969a3f2fe001790b0a1'
};
