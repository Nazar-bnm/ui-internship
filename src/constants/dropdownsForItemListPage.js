export const dropdownsForItemListPage = {
  itemsOnPage: [
    { value: '6', label: '6' },
    { value: '9', label: '9' },
    { value: '12', label: '12' }
  ],
  sortBy: [
    { value: 'Position', label: 'Position' },
    { value: 'Price', label: 'Price' },
    { value: 'Name', label: 'Name' }
  ]
};

// export default dropdownsForItemListPage;
