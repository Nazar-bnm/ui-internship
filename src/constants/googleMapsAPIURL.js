export const googleMapsGeocodeURL = 'https://maps.googleapis.com/maps/api/geocode';
