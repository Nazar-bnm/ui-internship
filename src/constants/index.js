export const defaultImages = [
  { src: 'src/assets/img/content/new4.png' },
  { src: 'src/assets/img/content/new2.png' },
  { src: 'src/assets/img/content/new3.png' },
  { src: 'src/assets/img/content/new1.png' }
];
