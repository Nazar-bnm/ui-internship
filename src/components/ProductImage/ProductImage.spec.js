import React from 'react';
import { shallow } from 'enzyme';
import ProductImage from './ProductImage';
import { defaultImages } from '../../constants';

describe('<ProductImage />', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(
      <ProductImage imagesArr={defaultImages} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
