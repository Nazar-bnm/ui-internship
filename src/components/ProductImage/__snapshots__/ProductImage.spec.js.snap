// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductImage /> should match snapshot 1`] = `
<div
  className="product-image__container content"
>
  <div
    className="product-image__big-image-container"
    type="submit"
  >
    <img
      alt="product"
      className="product-image"
      src="src/assets/img/content/new4.png"
    />
  </div>
  <div
    className="product-image__small-images-container"
  >
    <div
      className="product-image__small-image"
      key="src/assets/img/content/new4.png"
    >
      <img
        alt="product"
        className="product-image"
        src="src/assets/img/content/new4.png"
      />
      <button
        className="product-image__image-on-hover"
        data-src="src/assets/img/content/new4.png"
        label="clickable image"
        onClick={[Function]}
        type="submit"
      />
    </div>
    <div
      className="product-image__small-image"
      key="src/assets/img/content/new2.png"
    >
      <img
        alt="product"
        className="product-image"
        src="src/assets/img/content/new2.png"
      />
      <button
        className="product-image__image-on-hover"
        data-src="src/assets/img/content/new2.png"
        label="clickable image"
        onClick={[Function]}
        type="submit"
      />
    </div>
    <div
      className="product-image__small-image"
      key="src/assets/img/content/new3.png"
    >
      <img
        alt="product"
        className="product-image"
        src="src/assets/img/content/new3.png"
      />
      <button
        className="product-image__image-on-hover"
        data-src="src/assets/img/content/new3.png"
        label="clickable image"
        onClick={[Function]}
        type="submit"
      />
    </div>
  </div>
</div>
`;
